test/Cruzo1155Vault.ts

```diff
@@ -0,0 +1,300 @@
+import { SignerWithAddress } from "@nomiclabs/hardhat-ethers/signers";
+import { expect } from "chai";
+import { ethers, upgrades } from "hardhat";
+import { Contract } from "ethers";
+import { getEvent } from "../utils/getEvent";
+import { keccak256 } from "ethers/lib/utils";
+import { RAW_VAULT_FUNCTION_SIGNATURE, RAW_FACTORY_INITIALIZE_SIGNATURE } from "../constants/signatures"
+
+describe("CruzoVault", () => {
+    let market: Contract;
+    let beacon: Contract;
+    let factory: Contract;
+    let token: Contract;
+    let vault: Contract;
+    let tokenV2: Contract;
+    let owner: SignerWithAddress;
+    let seller: SignerWithAddress;
+    let buyer: SignerWithAddress;
+    let claimer: SignerWithAddress;
+    let claimer2: SignerWithAddress;
+
+
+    const serviceFee = 300;
+    const rawVaultFuncSignature = RAW_VAULT_FUNCTION_SIGNATURE
+    const tokenDetails = {
+        name: "Cruzo",
+        symbol: "CRZ",
+        baseOnlyURI: "https://cruzo.market",
+        baseAndIdURI: "https://cruzo.io/tokens",
+        altBaseOnlyURI: "https://opensea.io/tokens/{id}.json",
+        ipfsHash: "Qme3TrFkt28tLgHR2QXjH1ArfamtpkVsgMc9asdw3LXn7y",
+        altBaseAndIdURI: "https:opensea.io/tokens/",
+        collectionURI: "https://cruzo.io/collection",
+    };
+
+    beforeEach(async () => {
+        [owner, seller, buyer, claimer, claimer2] = await ethers.getSigners();
+
+        const CruzoMarket = await ethers.getContractFactory("CruzoMarket");
+        const Cruzo1155 = await ethers.getContractFactory("Cruzo1155");
+        const Factory = await ethers.getContractFactory("Cruzo1155Factory");
+        const Vault = await ethers.getContractFactory("Cruzo1155Vault");
+
+        market = await upgrades.deployProxy(CruzoMarket, [serviceFee, rawVaultFuncSignature], {
+            kind: "uups",
+        });
+        await market.deployed();
+        beacon = await upgrades.deployBeacon(Cruzo1155);
+        await beacon.deployed();
+        factory = await Factory.deploy(
+            beacon.address,
+            RAW_FACTORY_INITIALIZE_SIGNATURE,
+            "https://cruzo.market",
+            market.address
+        );
+        await factory.deployed();
+        vault = await upgrades.deployProxy(Vault, [market.address], {
+            kind: "uups",
+        });
+        await vault.deployed();
+        await market.setVaultAddress(vault.address);
+    });
+
+    describe("Simple token gift without wallet", () => {
+        it("Give and claim a gift", async () => {
+            const tokenId = ethers.BigNumber.from("1");
+            const supply = ethers.BigNumber.from("100");
+            const tradeAmount = ethers.BigNumber.from("10");
+            const price = ethers.utils.parseEther("0.01");
+            const utf8Encode = new TextEncoder();
+            const secretKey = "skdjf87ddg9ddfud9d9fu"
```

Comment:
No need to copy-paste it 5 times. Also, you can use some pretty value like "secret" / "this_is_secret".